bundles/org.openhab.binding.elroconnects/src/main/java/org/openhab/binding/elroconnects/internal/devices/ElroConnectsDeviceEntrySensor.java

```diff
@@ -0,0 +1,106 @@
+/**
+ * Copyright (c) 2010-2021 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.elroconnects.internal.devices;
+
+import static org.openhab.binding.elroconnects.internal.ElroConnectsBindingConstants.*;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.binding.elroconnects.internal.ElroConnectsBindingConstants.ElroDeviceStatus;
+import org.openhab.binding.elroconnects.internal.handler.ElroConnectsBridgeHandler;
+import org.openhab.binding.elroconnects.internal.handler.ElroConnectsDeviceHandler;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.OpenClosedType;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.types.UnDefType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link ElroConnectsDeviceEntrySensor} is representing a generic Elro Connects Entry Sensor device.
+ *
+ * @author Mark Herwege - Initial contribution
+ */
+@NonNullByDefault
+public class ElroConnectsDeviceEntrySensor extends ElroConnectsDevice {
+
+    private final Logger logger = LoggerFactory.getLogger(ElroConnectsDeviceEntrySensor.class);
+
+    // device states
+    private static final String STAT_OPEN = "55";
+    private static final String STAT_STILL_OPEN = "66";
+    private static final String STAT_FAULT = "11";
+    private static final String STAT_NORMAL = "AA";
```

Comment:
You are using the same constants in multiple classes. Consider moving them into a single static class so they can be re-used in all classes. When you make something static it will use RAM/memory even if you have no instances of the class, so by doing it multiple times it will use more ram then a single class that you re-use.

Reply:
The problem is that each of the devices may have different codes for these. Many overlap, but not all. That's why they have been separated out. That way, the codes for one device type are kept together.

Reply:
No problem leave as is and I will close this.